AfroBench: How Good are Large Language Models on African Languages?

JessicaOjo opened the pull request:
## Overview

AfroBench is a large scale benchmark of African languages on LLMs, covering 15 tasks and 22 datasets. 

### Features
The PR features 19 new tasks/datasets all in the AfroBench subfolder
It also updates the irokobench tasks - AfriXNLI, AfriMGSM and AfriMMLU to reflect recent experiments performed on these tasks

### Testing
The PR has been tested locally against several open source models

### Documentation
Adequate documentation has been provided on each task ReadMe pages.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2825) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2825) before we can accept your contribution.<br/>**1** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: JessicaOjo<br/>:x: dadelani<br/>:x: theyorubayesian<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=2825) it.</sub>

baberabb commented:
Hi! Thanks for your substantial PR. I'll try to review it soon! Just need everyone in the commit history to pacify the CLI assistant by signing the agreement. cc: @theyorubayesian, @dadelani

JessicaOjo commented:
> Hi! Thanks for your substantial PR. I'll try to review it soon! Just need everyone in the commit history to pacify the CLI assistant by signing the agreement. cc: @theyorubayesian, @dadelani

Thanks for the comment everyone in the commit history has signed the agreement.

StellaAthena commented:
> > Hi! Thanks for your substantial PR. I'll try to review it soon! Just need everyone in the commit history to pacify the CLI assistant by signing the agreement. cc: @theyorubayesian, @dadelani
> 
> Thanks for the comment everyone in the commit history has signed the agreement.

It looks like a bunch of tests are failing. Can you run the pre-commit script and see about addressing those issues?

JessicaOjo commented:
> > > Hi! Thanks for your substantial PR. I'll try to review it soon! Just need everyone in the commit history to pacify the CLI assistant by signing the agreement. cc: @theyorubayesian, @dadelani
> > 
> > 
> > Thanks for the comment everyone in the commit history has signed the agreement.
> 
> It looks like a bunch of tests are failing. Can you run the pre-commit script and see about addressing those issues?

Hi Stella,

I've addressed the pertinent issues, but I’m unsure if the remaining ones are within my scope to fix:
- Linters: The files are being formatted, but I’m not sure what’s causing the failure.
- External LM test: It’s failing due to an accelerate error—specifically, 'clear_device_cache' from accelerate.utils.memory.
- Scan for changed task test: This seems to be failing at the test script level.
Let me know if you have any suggestions on how I can resolve these. Thanks!

baberabb commented:
> Hi Stella,
> 
> I've addressed the pertinent issues, but I’m unsure if the remaining ones are within my scope to fix:
> 
> * Linters: The files are being formatted, but I’m not sure what’s causing the failure.
> * External LM test: It’s failing due to an accelerate error—specifically, 'clear_device_cache' from accelerate.utils.memory.
> * Scan for changed task test: This seems to be failing at the test script level.
>   Let me know if you have any suggestions on how I can resolve these. Thanks!

Hi!
Looks mostly great but could you run the pre-commit, I can't seem to push on my end

```bash
pip install pre-commit
pre-commit install
pre-commit run --all-files
```

JessicaOjo commented:
> > Hi Stella,
> > I've addressed the pertinent issues, but I’m unsure if the remaining ones are within my scope to fix:
> > 
> > * Linters: The files are being formatted, but I’m not sure what’s causing the failure.
> > * External LM test: It’s failing due to an accelerate error—specifically, 'clear_device_cache' from accelerate.utils.memory.
> > * Scan for changed task test: This seems to be failing at the test script level.
> >   Let me know if you have any suggestions on how I can resolve these. Thanks!
> 
> Hi! Looks mostly great but could you run the pre-commit, I can't seem to push on my end
> 
> ```shell
> pip install pre-commit
> pre-commit install
> pre-commit run --all-files
> ```
 
Hi! Thanks for the tip! I’ve run the pre-commit on all files and it passes locally.

Regarding the failed tests:
* Linters: The failure appears in a file untouched by this PR.
* Scan for changed tasks: This fails due to the use of a gated dataset (Sunbird/salt). Although access is simple with authentication, CI environments typically lack pre-approved access.

Let me know if there’s anything else I can do to help resolve these.

StellaAthena reviewed (approved):
Thank you for your contribution! We are excited to have this benchmark suported in the eval harness!
